lib/ansible/modules/windows/win_chocolatey.ps1

```diff
@@ -16,532 +19,543 @@ $params = Parse-Args $args -supports_check_mode $true
-            return
+        if ([Net.SecurityProtocolType].GetMember("Tls12").Count -gt 0) {
+            $security_protcols = $security_protcols -bor [Net.SecurityProtocolType]::Tls12
         }
-    }
-
-    $options = @( "-y", $package, "--timeout", "$timeout", "--failonunfound" )
-
-    switch ($architecture) {
-      "x86" { $options += "--x86" ; break}
-    }
-
-    if ($check_mode)
-    {
-        $options += "--whatif"
-    }
-
-    if ($version)
-    {
-        $options += "--version", $version
-    }
+        [Net.ServicePointManager]::SecurityProtocol = $security_protcols
 
-    if ($source)
-    {
-        $options += "--source", $source
-    }
+        $client = New-Object -TypeName System.Net.WebClient
+        $environment = @{}
+        if ($proxy_url) {
+            # the env values are used in the install.ps1 script when getting
+            # external dependencies
+            $environment.chocolateyProxyLocation = $proxy_url
+            $web_proxy = New-Object -TypeName System.Net.WebProxy -ArgumentList $proxy_url, $true
+            $client.Proxy = $web_proxy
+            if ($proxy_username -and $proxy_password) {
+                $environment.chocolateyProxyUser = $proxy_username
+                $environment.chocolateyProxyPassword = $proxy_password
+                $sec_proxy_password = ConvertTo-SecureString -String $proxy_password -AsPlainText -Force
+                $web_proxy.Credentials = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList $proxy_username, $sec_proxy_password
+            }
+        }
 
-    if ($force)
-    {
-        $options += "--force"
-    }
+        if ($source) {
+            # check if the URL already contains the path to install.ps1
+            if ($source.EndsWith("install.ps1")) {
+                $script_url = $source
+            } else {
+                # chocolatey server automatically serves a script at
+                # http://host/install.ps1, we rely on this behaviour when a
+                # user specifies the choco source URL. If a custom URL or file
+                # path is desired, they should use win_get_url/win_shell
+                # manually
+                # we need to strip the path off the URL and append install.ps1
+                $uri_info = [System.Uri]$source
+                $script_url = "$($uri_info.Scheme)/$($uri_info.Authority)/install.ps1"
+            }
+        } else {
+            $script_url = "https://chocolatey.org/install.ps1"
+        }
 
-    if ($installargs)
-    {
-        $options += "--installargs", $installargs
-    }
+        $install_script = $client.DownloadString($script_url)
+        if (-not $check_mode) {
+            $res = Run-Command -command "powershell.exe -" -stdin $install_script -environment $environment
+            if ($res.rc -ne 0) {
+                Fail-Json -obj $result -message "Chocolatey bootstrap installation failed. RC: $($res.rc), STDOUT: $($res.stdout), STDERR: $($res.stderr)"
```

Comment:
I would add stdout and stderr as keys in the result dictionary, as is custom with other modules.

They won't be part of the stable interface, but it would appear better (or could be accessed programmatically).

Reply:
Yep that makes sense, will update it